SerialPrograms/Source/PokemonSV/Options/PokemonSV_BBQOption.cpp

```diff
@@ -0,0 +1,103 @@
+/*  BBQ Option
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include <map>
+#include "PokemonSV_BBQOption.h"
+#include "PokemonSV/Inference/Boxes/PokemonSV_IvJudgeReader.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+BBQOption::BBQOption(OCR::LanguageOCROption* language_option)
+    : GroupOption(
+        "Blueberry Quests",
+        LockMode::UNLOCK_WHILE_RUNNING,
+        false, true
+    )
+    , m_language_owner(language_option == nullptr
+        ? new OCR::LanguageOCROption(
+            "<b>Game Language:</b><br>This is required to read quests.",
+            IV_READER().languages(),
+            LockMode::LOCK_WHILE_RUNNING,
+            true
+        )
+        : nullptr
+    )
+    , LANGUAGE(language_option == nullptr ? *m_language_owner : *language_option)
+    , NUM_QUESTS(
+        "<b>Number of Quests to run:</b>",
+        LockMode::UNLOCK_WHILE_RUNNING,
+        1000
+    )
+    , SAVE_NUM_QUESTS(
+        "<b>Save and reset the game after attempting this many quests:</b><br>This preserves progress and prevents potential game lags from long runs.<br>0 disables this option.",
+        LockMode::UNLOCK_WHILE_RUNNING, 50
+    )
+    , INVERTED_FLIGHT(
+        "<b>Inverted Flight:</b><br>Check this box if inverted flight controls are set.",
+        LockMode::LOCK_WHILE_RUNNING,
+        false
+    )
+    , QUICKBALL(
+        "<b>Catch Quest - Throw Quick Ball:</b><br>When attempting to catch a non-tera Pokemon, use a Quick Ball on the first turn.",
+        LockMode::UNLOCK_WHILE_RUNNING,
+        true
+    )
+    , BALL_SELECT(
+        "<b>Catch Quest - Ball Select:</b><br>What ball to use when performing a catch quest.",
+        LockMode::UNLOCK_WHILE_RUNNING,
+        "ultra-ball"
+    )
+    , NUM_EGGS(
+        "<b>Hatch Quest - Number of Eggs:</b><br>Amount of eggs located in your current box.",
+        LockMode::LOCK_WHILE_RUNNING, 30, 0, 30
+    )
+    , OUT_OF_EGGS(
+        "<b>Hatch Quest - Zero Eggs:</b><br>When out of eggs to hatch, do the selected option.",
+        {
+            {OOEggs::Stop,      "stop",         "Stop Program"},
+            {OOEggs::Reroll,    "reroll",       "Reroll Quest - Costs 50BP"},
+            {OOEggs::KeepGoing, "keep-going",   "Keep Going - Warning: Bonus(Red) quests will be blocked!"},
+        },
+        LockMode::LOCK_WHILE_RUNNING,
+        OOEggs::Stop
```

Comment:
Any reason not to do KeepGoing ? From my experience, it seems you can "stack up" red quest, meaining even if you do not complete the current red quest, you're already working on unlocking the next red quest

Reply:
It only stacks up an additional two red quests. The bar fills green->gold->red and then stops there.

![image](https://github.com/PokemonAutomation/Arduino-Source/assets/3212217/307644e5-5ded-459d-b4cd-9ca06d5dbf73)
![image](https://github.com/PokemonAutomation/Arduino-Source/assets/3212217/c681875d-b4e9-4669-a21f-18acc90f0557)